src/engine/datafacade/process_memory_allocator.cpp

```diff
@@ -15,20 +15,26 @@ ProcessMemoryAllocator::ProcessMemoryAllocator(const storage::StorageConfig &con
     storage::Storage storage(config);
 
     // Calculate the layout/size of the memory block
-    storage::DataLayout layout;
-    storage.PopulateStaticLayout(layout);
-    storage.PopulateUpdatableLayout(layout);
+    std::vector<std::pair<bool, boost::filesystem::path>> static_files = storage.GetStaticFiles();
+    std::vector<std::pair<bool, boost::filesystem::path>> updatable_files =
+        storage.GetUpdatableFiles();
+    std::unique_ptr<storage::BaseDataLayout> layout = std::make_unique<storage::DataLayout>();
+    storage.PopulateLayoutWithRTree(layout);
+    storage.PopulateLayout(layout, static_files);
+    storage.PopulateLayout(layout, updatable_files);
 
     // Allocate the memory block, then load data from files into it
-    internal_memory = std::make_unique<char[]>(layout.GetSizeOfLayout());
+    internal_memory = std::make_unique<char[]>(layout->GetSizeOfLayout());
 
-    index = storage::SharedDataIndex({{internal_memory.get(), std::move(layout)}});
+    std::vector<storage::SharedDataIndex::AllocatedRegion> regions;
+    regions.push_back({internal_memory.get(), std::move(layout)});
+    index = {std::move(regions)};
 
     storage.PopulateStaticData(index);
     storage.PopulateUpdatableData(index);
 }
 
-ProcessMemoryAllocator::~ProcessMemoryAllocator() {}
+ProcessMemoryAllocator::~ProcessMemoryAllocator() { /* free(internal_memory) */}
```

Comment:
I don't understand this - why is it commented out? What is this supposed to do? Do we need the empty dtor at all?

Reply:
Removed